src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -514,12 +528,14 @@ private TypeSpec CreateEnumerableSpec(TypeParseInfo typeParseInfo)
 
                 TypeRef elementTypeRef = EnqueueTransitiveType(typeParseInfo, elementType, DiagnosticDescriptors.ElementTypeNotSupported);
 
+                Debug.Assert(!shouldTestCast || !type.IsValueType, "Should not test cast for value types.");
```

Comment:
> Debug.Assert(!shouldTestCast || !type.IsValueType, "Should not test cast for value types."); [](http://example.com/codeflow?start=15&length=93)

similar to previous question

Reply:
Same answer - the test down cast is only applicable when the bound type is itself `IReadonlyDictionary`, not for types that implement `IReadonlyDictionary`.